--- a/libraw/libraw_datastream.h
+++ b/libraw/libraw_datastream.h
@@ -43,6 +43,10 @@
 #endif
 #endif
 
+#if defined(__ANDROID__) && __ANDROID_API__ < 24
+#define _LIBCPP_HAS_NO_OFF_T_FUNCTIONS
+#endif
+
 #include "libraw_const.h"
 #include "libraw_types.h"
 #include <fstream>
--- a/src/libraw_datastream.cpp
+++ b/src/libraw_datastream.cpp
@@ -588,6 +588,8 @@
 #else
   return fseek(f, (long)o, whence);
 #endif
+#elif __ANDROID__ && __ANDROID_API__ < 24
+  return fseek(f, (long)o, whence);
 #else
   return fseeko(f, o, whence);
 #endif
@@ -602,6 +604,8 @@
 #else
   return ftell(f);
 #endif
+#elif __ANDROID__ && __ANDROID_API__ < 24
+  return ftell(f);
 #else
   return ftello(f);
 #endif
